flycheck.el

```diff
@@ -8621,8 +8621,10 @@ for more information about the custom directories."
 (defun flycheck-eslint-config-exists-p ()
   "Whether there is a valid eslint config for the current buffer."
   (let* ((executable (flycheck-find-checker-executable 'javascript-eslint))
-         (exitcode (and executable (call-process executable nil nil nil
-                                                 "--print-config" "."))))
+         (exitcode (and executable
+                        (call-process executable nil nil nil
+                                      "--print-config" (or buffer-file-name
+                                                           "index.js")))))
```

Comment:
Will passing index.js actually work? If we don't have a buffer-file-name, how will eslint know which file we're checking when we actually call it? (source-original will return an empty string)

Reply:
I agree with @cpitclaudel. I think checking for `buffer-file-name` is enough.

Reply:
So if you don't have a backing file, `print-config` will fail and the checker can't be used?

Reply:
In which case do you expect flycheck-eslint to be enabled but have no backing file to send to eslint itself?

Reply:
We are piping the buffer via stdin to eslint, so I think the checker currently works without a backing file (using the default configuration for the working directory).

Not sure if it's used by many people though.

Reply:
I use it :) eslint has become pretty essential for JS-development for many.

Reply:
@cjk really? We're not talking about a buffer not yet saved to disk; rather,  we're discussing a buffer that doesn't even have a buffer-file-name.  
Also, what happens if we casll eslint with a non-existing file?

Reply:
> Also, what happens if we call eslint with a non-existing file?

```session
$ eslint foo.js

Oops! Something went wrong! :(

ESLint: 6.0.1.

No files matching the pattern "foo.js" were found.
Please check for typing mistakes in the pattern.

$ echo $?
2

$ eslint --print-config foo.js
{
  "env": { ... }
}

$ echo $?
0
```

Reply:
OK.  Let's just use index.js then? I think that's fine by me.

Reply:
> OK. Let's just use index.js then? I think that's fine by me.

I suggest to keep `(or buffer-file-name "index.js")` because there might be some filename-specific configuration options. For example, files matching `tests-*.js` can have a different set of eslint rules.

Reply:
yes yes of course, that's what I meant; sorry for being unclear ^^

Reply:
All in agreement then.  Thank you for being thorough, everyone.

Reply:
@cpitclaudel You're right, not using it on unbound buffers. Sorry for the confusion.